bootstrap.hs

```diff
@@ -0,0 +1,35 @@
+import Control.Monad
+import Data.Foldable
+import System.FilePath
+import System.Directory
+import System.Process
+import System.Exit
+
+main :: IO ()
+main = do
+  let bootstrap_root = "./bootstrap-root"
+      happy_boot = bootstrap_root </> "happy"
+  callProcess "cabal"
+    [ "install", "happy",
+      "-f", "-bootstrap",
```

Comment:
```suggestion
      "-f", "+bootstrap",
```
Don't we want that?

Reply:
That would require a pre-installed `happy`

Reply:
No we don't

Reply:
With `-f -bootstrap` we don't, with `-f +bootstrap` we do. That's how this flag works, and I think you named it 😄

Reply:
Sorry! I wasn't loading your comments, I meant to reply to myself not rudely reply to you! :flushed: :grimacing:

Reply:
Yes, this is not the first time my own name for the flag has confused me :sweat_smile:

Reply:
I think this also confuses me.  
I would think that `-f +bootstrap` would do something so that _this_ instance of `happy` can build _itself_, whereas a `-f -bootstrap` would require _another_ (already installed) instance of `happy`.
Or in other words, _flag on_ does something special, _flag off_ does the vanilla thing.